### ISSUE TITLE
"Artifactory response: 400 Bad Request" since 1.9.4

### ISSUE THREAD

**Issue by lindhe** [jfrog/artifactory-azure-devops-extension#158]
Since yesterday, our CI pipeline have been failing when pushing Docker images to Artifactory. We see that this occurred at the same time as this task was upgraded from 1.9.2 to 1.9.4. Posting this here mostly in case anyone else has the same problem.

**Comment by Or-Gabay** [comment]
@lindhe, could you please provide more details about your issue?

**Comment by eyalbe4** [comment]
@lindhe,
We're here to help, but we do need more information about the issue you're reporting. Can you please share with us the build logs? What exactly is your task doing?

**Comment by lindhe** [comment]
Sorry for the initial lack of details. The last two days have been very intense for us, trying to debug this. We've had two support calls with JFrog and a team of 6 people debugging things all day. We think we have figured out what the issue is (but there's no way to know for sure, for reasons I'll get into later).

We used the `ArtifactoryDocker` task slightly incorrectly, by providing a `targetRepo` that didn't exist in our Artifactory. However, we did tag the images correctly, so the push still went through. My guess is that the push caused an error, but that the task never caught that error. So with this new version, the error was caught and it started failing our pipeline. Both last week and earlier today during the breakage, all images were still pushed to the intended repo in Artifactory, that's why I'm suspecting that we used to get an error but it was never made visible to us before today.

So our code, having been unchanged and working for months, suddenly stopped working because you pushed this patch version. While patch versions are supposed to not cause behavioral changes, I can totally see why this happened. First of all, if my theory is right, it sounds like the old behavior is incorrect and this is a bugfix. Secondly, mistakes happens and no-matter if this was a mistake from your side or not, our pipeline should be able to handle mistakes.

The real problem here is that [Microsoft provides us no way to pin the exact version number](https://developercommunity.visualstudio.com/content/problem/1036144/cannot-pin-task-version-making-pipelines-completel.html) of a task. So there's no way for us to try out `1.9.2` to verify our theory. Nor can we pin the version to ensure that breaking changes like this are not rolled out automatically (and irreversibly).

Right now, until Microsoft gets their act together and figures out a method that let's us choose a specific version of a task **(like every other sane package manager in existence, mind you)**, every new change for the tasks we use in our pipeline might cause breaking changes for us.

I should also add that this is the [second time](https://github.com/jfrog/artifactory-azure-devops-extension/issues/149) in a month this happens to us, so it's not a one-timer that can be shrugged off. So therefore, we hope to collaborate with you to find a work-around until Microsoft implements a real long-term solution.

My main reason for putting a ticket here too was that maybe someone else had the same issue, and would be able to find information about it now. Marking this as closed, and we'll continue this journey via the ticket we have on the JFrog enterprise support forums. They said that they would get us in contact with the task developers, which I assume means you guys. So we'll talk soon, hopefully. :)
